### PR TITLE
feat: export chats per channel

### LINKED ISSUE #2077: User should be able to export chats
A user requested the ability to export all chats to flat text files for each channel, e.g. 

```
[alice 20:23]
Hi Bob!

[bob 20:24]
Sup Alice!
```

We should make this possible in the simplest possible way. We could add an "Export chats" item to settings, for example. Or have a command-line option, or even just a script that does this. The user is technical (and based on existing user requests I think it's reasonable to expect that users who want this initially will be somewhat technical or tolerant of technical steps) so we can do whatever is simplest.

### PR DESCRIPTION
**Pull request by rajdip-b**
Fixes #2077 

### Pull Request Checklist

- [x] I have linked this PR to related GitHub issue.
- [x] I have updated the CHANGELOG.md file with relevant changes (the file is located at the root of monorepo).

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [x] I have updated base screenshots for visual regression tests

Features introduced: 
- Every channel option now has a **Export chats** option to export chats
image

- Chats are exported into a text file of the user's preference in this format:
image

### CODE REVIEW

**Comment by rajdip-b** [comment]
@holmesworcester Can I get a review on this?

**Comment by holmesworcester** [comment]
Looks great! Thank you!!!

I just approved the workflow to run tests. I think some of the snapshot tests will be broken by your change so we'll have to update those I bet.

Also I think it would be good to add a couple tests to make sure that the UI and text file generation works correctly. If you don't have time for writing tests I think that's okay in this case and we're grateful for the help! But it would be ideal to have some.

@Kacper-RF what was the workflow for letting new contributors update visual regression tests? Are there any tests you would suggest here?

I'd call the menu item 'Export messages' in the UI since we don't use or plan to use 'chats' elsewhere in the UI.

**Comment by rajdip-b** [comment]
> Looks great! Thank you!!!
> 
> I just approved the workflow to run tests. I think some of the snapshot tests will be broken by your change so we'll have to update those I bet.
> 
> Also I think it would be good to add a couple tests to make sure that the UI and text file generation works correctly. If you don't have time for writing tests I think that's okay in this case and we're grateful for the help! But it would be ideal to have some.
> 
> @Kacper-RF what was the workflow for letting new contributors update visual regression tests? Are there any tests you would suggest here?
> 
> I'd call the menu item 'Export messages' in the UI since we don't use or plan to use 'chats' elsewhere in the UI.

I will be absolutely fine with writing the tests and making the tweaks you mentioned. I will be needing a few references though for the test since I'm new to react testing :(

File: packages/desktop/src/renderer/components/ContextMenu/menus/ChannelContextMenu.container.tsx [hunk 1]
```diff
@@ -10,10 +10,12 @@ import { ContextMenuItemProps } from '../ContextMenu.types'
 
 import { useModal } from '../../../containers/hooks'
 import { ModalName } from '../../../sagas/modals/modals.types'
+import { exportChats } from 'packages/desktop/src/utils/functions/exportChats'
```
Comment (Kacper-RF) [inline packages/desktop/src/renderer/components/ContextMenu/menus/ChannelContextMenu.container.tsx:13]: Please fix this import, this is the reason why all e2e tests and one desktop test failed.

Comment (rajdip-b) [reply packages/desktop/src/renderer/components/ContextMenu/menus/ChannelContextMenu.container.tsx:13]: Sure thing! I'll do this right away.

**Comment by Kacper-RF** [comment]
> Looks great! Thank you!!!
> 
> I just approved the workflow to run tests. I think some of the snapshot tests will be broken by your change so we'll have to update those I bet.
> 
> Also I think it would be good to add a couple tests to make sure that the UI and text file generation works correctly. If you don't have time for writing tests I think that's okay in this case and we're grateful for the help! But it would be ideal to have some.
> 
> @Kacper-RF what was the workflow for letting new contributors update visual regression tests? Are there any tests you would suggest here?
> 
> I'd call the menu item 'Export messages' in the UI since we don't use or plan to use 'chats' elsewhere in the UI.

I remember fixing "Desktop Visual Regressions/Chromatic Deployment" to run by other authors and in this case it works.

The problem I see is that there is an incorrect import in one of the files and that is why some of the tests failed.

I would suggest maybe some component/snapshot test. @siepra Do you think something more is needed here?

**Comment by Kacper-RF** [comment]
@rajdip-b Thanks for your contribution, I left one comment that should help with tests.
For future please remember to merge newest develop branch into your branch before pushing commits (to prevent download alpha version with higher number by autoupdater on e2e workflow)

**Comment by rajdip-b** [comment]
> @rajdip-b Thanks for your contribution, I left one comment that should help with tests. For future please remember to merge newest develop branch into your branch before pushing commits (to prevent download alpha version with higher number by autoupdater on e2e workflow)

Thanks! I'll keep that in mind.

**Comment by rajdip-b** [comment]
Hey @Kacper-RF, I think the checks are passing now. I am trying to build a few tests for my changes, and then I will push the updated code.

**Comment by rajdip-b** [comment]
@Kacper-RF @holmesworcester can you give it a final review?

**Review by holmesworcester — COMMENTED** [review]
Looks good to me. @leblowl?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 * Fix: dial new peers on CSRs replication
 
+* Feature: add functionality to export chat to text document in desktop version
```
Comment (EmiM) [inline CHANGELOG.md:11]: Could you add this info under [unreleased] on top of the changelog? I think first you have to merge develop into your branch because we released a new alpha version today

Comment (rajdip-b) [reply CHANGELOG.md:11]: Oh, okay! I merged the develop branch just now. I'll check once more.

Comment (rajdip-b) [reply CHANGELOG.md]: @EmiM done!

**Comment by leblowl** [comment]
Thanks @rajdip-b! Overall looks good to me. I was testing it a bit and noticed the formatting in this case:
screen
results in a text file like:
```

[b 9:13 AM]
@b created #general

[b 9:14 AM]
test,[b 9:14 AM]
again,[b 9:15 AM]
yo
```
for me at least.

Of course being able to export chats at all is really nice, but perhaps it would help with readability to have each message on it's own line. What do you think?

Another thought is just getting the lint step and unit tests passing.

**Comment by rajdip-b** [comment]
> Thanks @rajdip-b! Overall looks good to me. I was testing it a bit and noticed the formatting in this case: screen results in a text file like:
> 
> ```
> 
> [b 9:13 AM]
> @b created #general
> 
> [b 9:14 AM]
> test,[b 9:14 AM]
> again,[b 9:15 AM]
> yo
> ```
> 
> for me at least.
> 
> Of course being able to export chats at all is really nice, but perhaps it would help with readability to have each message on it's own line. What do you think?
> 
> Another thought is just getting the lint step and unit tests passing. Maybe the desktop tests are failing due to obsolete snapshots?

That's really strange.... Let me test this thing again, since it worked earlier.

**Comment by rajdip-b** [comment]
@leblowl I tried the edge case you sent. It indeed was breaking! I have fixed the error. Do let me know in case you folks find any other issue.

**Comment by leblowl** [comment]
> @leblowl I tried the edge case you sent. It indeed was breaking! I have fixed the error. Do let me know in case you folks find any other issue.

@rajdip-b awesome, looks good! It looks like the `check` step is still failing due to 2 lint errors, do you think you'd be able to fix that up?

**Comment by rajdip-b** [comment]
Hey @leblowl, I informed this to @holmesworcester as well. I really have no idea about why that lint is failing since I never interacted with the file. Also, I couldn't fine any `prettier --fix` script in any of the `package.json` files. Please help me with this one :(

**Comment by leblowl** [comment]
> Hey @leblowl, I informed this to @holmesworcester as well. I really have no idea about why that lint is failing since I never interacted with the file. Also, I couldn't fine any `prettier --fix` script in any of the `package.json` files. Please help me with this one :(

Ok got it. I think we can just merge it then and figure out that error in another commit. @Kacper-RF @EmiM anything else that would like to see resolved?

**Review by leblowl — APPROVED** [review]
